### PR TITLE
[DOCS] Add retrievers overview

### PR DESCRIPTION
**Pull request by leemthompo**
## [URL preview](https://elasticsearch_bk_107959.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/retrievers-overview.html)

### CODE REVIEW

File: docs/reference/search/search-your-data/retrievers-overview.asciidoc [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+enhancing document retrieval and ranking logic.
+* *Combiners*. Compound retrievers that merge top hits 
+from multiple sub-retrievers. 
+//* NOT YET *Rerankers*. Special compound retrievers that reorder hits and may adjust the number of hits, with distinctions between first-stage and second-stage rerankers.
+
+[discrete]
+[[retrievers-overview-play-in-search]]
+=== Retrievers in action
+
+//Playground will be renamed
```
Comment (leemthompo) [inline docs/reference/search/search-your-data/retrievers-overview.asciidoc:121]:
```suggestion
```

File: docs/reference/search/search-your-data/retrievers-overview.asciidoc [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+[discrete]
+[[retrievers-overview-play-in-search]]
+=== Retrievers in action
+
+The Search Playground, builds Elasticsearch queries using the retriever abstraction.
+It automatically detects the fields and types in your index and builds a retriever tree based on your selections.
+
+You can use the Playground to experiment with different retriever configurations and see how they affect search results.
+
+Refer to the {kibana-ref}/playground.html[[Playground] documentation] for more information.
```
Comment (leemthompo) [inline docs/reference/search/search-your-data/retrievers-overview.asciidoc:207]:
```suggestion
Refer to the {kibana-ref}/playground.html[Playground documentation] for more information.
```

File: docs/reference/search/search-your-data/retrievers-overview.asciidoc [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+See the <<rrf-using-multiple-standard-retrievers,RRF documentation>> for detailed
+examples and information on how to use the RRF retriever.
+
+[NOTE]
+====
+Stay tuned for more retriever types in future releases!
+====
+
+[discrete]
+=== What Makes Retrievers Useful?
```
Comment (demjened) [inline docs/reference/search/search-your-data/retrievers-overview.asciidoc:51]: Nit: consistency with capitalization of other titles

```suggestion
=== What makes retrievers useful? 
```

File: docs/reference/search/search-your-data/retrievers-overview.asciidoc [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+enhancing document retrieval and ranking logic.
+* *Combiners*. Compound retrievers that merge top hits 
+from multiple sub-retrievers. 
+//* NOT YET *Rerankers*. Special compound retrievers that reorder hits and may adjust the number of hits, with distinctions between first-stage and second-stage rerankers.
+
+[discrete]
+[[retrievers-overview-play-in-search]]
+=== Retrievers in action
+
+The Search Playground, builds Elasticsearch queries using the retriever abstraction.
```
Comment (demjened) [inline docs/reference/search/search-your-data/retrievers-overview.asciidoc:202]:
```suggestion
The Search Playground builds Elasticsearch queries using the retriever abstraction.
```

File: docs/reference/search/search-your-data/retrievers-overview.asciidoc [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+
+.*Expand* for example
+[%collapsible]
+==============
+
+[source,js]
+----
+GET example-index/_search
+{
+    "sub_searches": [
```
Comment (demjened) [inline docs/reference/search/search-your-data/retrievers-overview.asciidoc:151]: Nit: consider using the same indentation as in the previous example (2 vs 4 spaces).

File: docs/reference/search/search-your-data/retrievers-overview.asciidoc [hunk 6]
```diff
@@ -0,0 +1,211 @@
...
+        {
+            "query": {
+                "term": {
+                    "text": "blue shoes sale"
```
Comment (demjened) [inline docs/reference/search/search-your-data/retrievers-overview.asciidoc:155]: Consider using the same query claues in both examples - `match` for BM25, `model_id: .elser_model_2` on the `vector.tokens` field for kNN.

**Review by demjened — COMMENTED** [review]
Nice doc, I left some minor comments.

File: docs/reference/search/search-your-data/retrievers-overview.asciidoc [hunk 7]
```diff
@@ -0,0 +1,211 @@
...
+
+* <<standard-retriever,*Standard Retriever*>>. Returns top documents from a 
+traditional https://www.elastic.co/guide/en/elasticsearch/reference/master/query-dsl.html[query]. 
+Mimics a traditional query but in the context of a retriever framework. This 
+ensures backward compatibility as existing `_search` requests remain supported. 
+That way you can transition to the new abstraction at your own pace without 
+mixing syntaxes.
+* <<knn-retriever,*kNN Retriever*>>. Returns top documents from a <<search-api-knn,knn search>>, 
+in the context of a retriever framework.
+* <<rrf-retriever,*RRF Retriever*>>. Combines and ranks multiple standard retrievers using
```
[the remark this answers is not in the document]
Comment (leemthompo) [reply docs/reference/search/search-your-data/retrievers-overview.asciidoc:34]: That's a good one for the glossary, let me know if there's other terms you think should be in there :)

**Comment by leemthompo** [comment]
We'll definitely update this based on feedback, but probably worth considering merging at this stage

**Review by demjened — APPROVED** [review]
LGTM

**Pull request merged**